crates/router/src/core/payments/tokenization.rs

```diff
@@ -779,6 +894,55 @@ pub async fn save_in_locker(
     }
 }
 
+#[cfg(all(feature = "v2", feature = "payment_methods_v2"))]
+pub async fn save_in_locker(
+    state: &SessionState,
+    merchant_account: &domain::MerchantAccount,
+    payment_method_request: api::PaymentMethodCreate,
+) -> RouterResult<(
+    api_models::payment_methods::PaymentMethodResponse,
+    Option<payment_methods::transformers::DataDuplicationCheck>,
+)> {
+    payment_method_request.validate()?;
+    let merchant_id = merchant_account.get_id();
+    let customer_id = payment_method_request
+        .customer_id
+        .clone()
+        .get_required_value("customer_id")?;
+    match payment_method_request.payment_method_data.clone() {
+        Some(api::PaymentMethodCreateData::Card(card)) => {
+            Box::pin(payment_methods::cards::add_card_to_locker(
+                state,
+                payment_method_request,
+                &card,
+                &customer_id,
+                merchant_account,
+                None,
+            ))
+            .await
+            .change_context(errors::ApiErrorResponse::InternalServerError)
+            .attach_printable("Add Card Failed")
+        }
+        _ => {
+            let pm_id = common_utils::generate_id(consts::ID_LENGTH, "pm");
```

Comment:
this function should not create payment_method_id, payment_method needs to be passed here.